### PR TITLE
Add support for structure config yaml file

### PR DESCRIPTION
**Pull request by zachgiordano**
* Add support for structure config yaml file
  * Specify the `structure-config-file` location in the register Structure command
  * Add ability to run structure from the CLI
  * Add structure_config pydantic model to Structure

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,118 +6,131 @@
...
-    1. Make sure you're logged in to GitHub.
-    2. Go to the [Managed Structure Template repo](https://github.com/griptape-ai/managed-structure-template).
-    3. Select the *Use this template* drop-down.
-    4. Choose *Create a new repository*.
-    5. Provide a name for the new repository and (optionally) a description.
-    6. Press the *Create repository* button.
-    7. You now have a repository in your own GitHub account that is a copy of the Managed Structure Template for you to begin working with.
+   1. Make sure you're logged in to GitHub.
+   2. Go to the [Managed Structure Template repo](https://github.com/griptape-ai/managed-structure-template).
+   3. Select the _Use this template_ drop-down.
```
Comment (SavagePencil) [inline README.md:49]: why italics vs. bold?

File: README.md [hunk 2]
```diff
@@ -6,118 +6,131 @@
...
+   Or you can register and set the environment variable in one step.
+
+   ```bash
+   export GT_STRUCTURE_ID=$(gt skatepark register --structure-config-file structure_config.yaml --tldr)
+   ```
 
 > [!IMPORTANT]
-> Structures registered with the Skatepark are not persisted across restarts. You will need to re-register the Structure each time you restart Skatepark.
-6. Confirm that the Structure is registered.
+> Structures registered with the Skatepark are not persisted across restarts. You will need to re-register the Structure each time you restart Skatepark. 6. Confirm that the Structure is registered.
```
Comment (SavagePencil) [inline README.md:82]: does this need a newline before the `6.`?

File: README.md [hunk 3]
```diff
@@ -6,118 +6,131 @@
...
+   1. Create a file named `.env` in the same directory that your Structure code is in.
+   2. Open the `.env` file in a text editor.
+   3. Copy the contents of `.env.structure.example` into the `.env` file. This specifies all of the values necessary and explains how they are used.
+   4. The template expects an `OPENAI_API_KEY` environment variable by default to function. Add OPENAI*API_KEY=\_your OpenAI API Key here* to the `.env` file and save it.
+   5. As you expand on the template, you may add any other environment variables your Structure depends on to this file.
+8. Rebuild the structure to load in the new environment variable.
+   Note that this is only required for changes to `.env` or `requirements.txt`. Code changes do not require a rebuild.
+   ```bash
+   gt skatepark build
+   ```
```
Comment (SavagePencil) [inline README.md:99]: did anything actually change here or is this just the formatter?

**Review by SavagePencil — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (zachgiordano) [reply README.md]: This was a bad auto-formatter. Fixed it back to bold.

File: README.md (hunk 2 above)
Comment (zachgiordano) [reply README.md]: Yes! Fixed this

File: README.md (hunk 3 above)
Comment (zachgiordano) [reply README.md]: Just the formatter here, and also a markdown linter extension

**Review by SavagePencil — APPROVED** [review]

**Review by collindutter — APPROVED** [review]

**Pull request merged**